### PR TITLE
Feature/api key validation

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/44827

This PR fixes a small uninitialised var issue and adds a validation method

### CODE REVIEW

File: src/Tribe/Validate.php [hunk 1]
```diff
@@ -167,6 +167,22 @@ public function alpha_numeric_multi_line_with_dots_and_dashes() {
...
+		/**
+		 * Validates a field as a string containing only letters,
+		 * numbers, dashes and underscores
+		 *
+		 * @return stdClass validation result object
+		 */
+		public function alpha_numeric_with_dashes_and_underscores() {
+			if ( preg_match( '/^[a-zA-Z0-9_-]+$/', $this->value ) ) {
+				$this->result->valid = true;
+				$this->value         = trim( $this->value );
```
Comment (barryhughes) [inline src/Tribe/Validate.php]: Given the pattern that's been matched, any need to use `trim()` here?

**Comment by barryhughes** [comment]
Looks good!

File: src/Tribe/Validate.php (hunk 1 above)
Comment (lucatume) [inline src/Tribe/Validate.php]: No, I've moved the trimming before it.

**Comment by barryhughes** [comment]
Thanks for making the change, good to merge :+1:

**Pull request merged**